### ISSUE TITLE
Reveal Not Working

### ISSUE THREAD

**Issue by Laramie43ip** [ethereum/ens-registrar-dapp#128]
Hi All:

I'm encountering some significant Reveal issues with dozens of bids.  Here is what I've done:

1. Submitted bid (for example, "cathaypacific."  

2. Transaction hash showing successful bid is here: https://etherscan.io/tx/0x60f0aaf81745a5a3273f571e86bccacbc17a64bdd7534f0452f1d03bb89e62cb.  The amount I bid, including anonymous amount, is correct.

3. I downloaded the .JSON file with the bid.  I can see the name of the bid in the file, and I can re-upload that bid into the browser.  Browser messages "X number of bids successfully imported."

4. Still, while using the Registrar DAPP, no Reveal option is presented.  I have confirmed that I am logged into Metamask and that the account I used to bid is correct.

This is happening with multiple bids placed yesterday.  Also, the DAPP itself is incredibly slow and buggy today.  I'm using Metamask through Google Chrome under Mac OS.  Any suggestions?  Thanks in advance.

**Comment by Arachnid** [comment]
Can you confirm the hash of the name you're bidding on in that example transaction, and when the reveal period started?

**Comment by Laramie43ip** [comment]
If you mean the transaction hash of the bid for cathaypacific, it's here: 0x60f0aaf81745a5a3273f571e86bccacbc17a64bdd7534f0452f1d03bb89e62cb

Reveal period start was 2017-05-11 17:03

Here's another example: 
Name = smarkets
Bid URL: https://etherscan.io/tx/0xf093b52406eccee3286db0fae92cb7db00608e6b92735885e5cd7337c7811c04
Bid hash: 0xf093b52406eccee3286db0fae92cb7db00608e6b92735885e5cd7337c7811c04
Reveal Period Commenced: 2017-05-11 17:03

I have the bid confirm in a .JSON file.  I can successfully import that file.  Still, no opportunity to Reveal is presented.

I have been downloading after each successful bid.  The download file invariably does not include every name that I have bid on.  It usually takes 10 minutes or so after a successful bid for a name to appear.  It also appears that older names are being eliminated.  If you open many of the .JSON downloads, you can usually find a file with the bid name.  But, even uploading this file with a successful import confirmation isn't working at the moment.

Thanks for the help.

**Comment by Laramie43ip** [comment]
Or, do you need what appears to be the name hash from the .JSON file?

**Comment by alexvandesande** [comment]
Can you please open the developer console for the ens app in mist and paste the following info:

```
MyBids.findOne({name:'cathaypacific'});
```

And tell me what you see? If your bid is saved, then it should be there. If not, you can see all your bids by looking at 

```
MyBids.find().fetch();
```

Worst case scenario, you should be able to use MEW to reveal it..

**Comment by Laramie43ip** [comment]
I'm working from a Chrome browser using Metamask.  I've got the Dev console open in Chrome.  There is a section in console titled "update Reveal Names" and neither of these are there.  But, I'm not a Dev so I'm probably not looking in the correct place.

Is it possible to Reveal Names in MEW by simply uploading the pertinent .JSON file I downloaded in Chrome?  Thanks again for the help.

**Comment by pmandel24** [comment]
I ran into this issue as well on Mist, Mac Sierra.  Reveal bid button never showed, but the dapp showed the second highest bid amount as my bid amount which led me to believe that maybe the Dapp auto-revealed for me.  

This DAPP needs a complete re-write of the GUI. It was impossible to determine if the app wasn't functioning correctly or if I was missing a step. It's also impossible to determine where my active bids are supposed to show up, or how to check on an active bid, so I have no feedback or indication if I successfully placed a bid or where to check on it. 

The steps of the process should all be visible but grayed out rather than disappearing. For a new user it's confusing as to what's supposed to happen next. 

Pretty frustrating, burned 0.5 ether because of this bug and the bad UI on the dapp. 

The DAPP also freezes for 3-5 minutes while loading/syncing the first time. Probably because it's fetching the KNOWNS AUCTIONS STARTED RECENTLY and LAST CHANCE TO BID! lists. These should be lazy-loaded or on separate screens so the initial app doesn't freeze. I'm not up to speed on DAPP gui design constraints, but I would assume there's a technique to run the GUI asynchronously from the ether network updates.

**Comment by Laramie43ip** [comment]
I would suggest that anyone bidding in ENS move away from the DAPP/Metamask and over to MyEtherWallet.  There are some serious issues with bidding using Chrome/Metamask.  Bids get confirmed.  You get a confirming transaction hash.  You can go to Etherscan and see that the bid went through.  Funds were deducted from your wallet.  But, the bid doesn't get stored in the browser or in the downloaded .JSON file.  When you go to Reveal, the browser does not have the necessary data to prompt the Reveal option.  Your bid then expires and your funds are burned.  

The two issues I've seen are: (a) some confirmed bids never populate in the browser or the download file, and (b) some bids show up but only after significant (> 1 day) delay.  Downloading your bids is not foolproof.  It does help, because you can occasionally find confirmed bids in an earlier .JSON file even if they disappear in a subsequent .JSON bid download file.  

The speed issues are another reason to go to MEW.

**Comment by Laramie43ip** [comment]
Also, switching between http and https helps when a bid is missing.  But, again, that also is not foolproof.  Some confirmed bids do not show up regardless of whether you use http or https.

**Comment by ShawnKimble** [comment]
Switching to https resolved my similar issue of bids dropping out of Chrome/Metamask.

**Comment by tcraigscott** [comment]
This is brutal, may have just lost a significant bit of ether due to this, bid went through but doesn't show up in metamask, and it froze before I got the chance to download the backup JSON file. Is there any way to reveal without the "secret" portion. It is clear the transaction came from my metamask wallet on etherscan, and I can prove access to that wallet.

**Comment by Arachnid** [comment]
No, there is no way to reveal without the secret.

If your bid does not show up in metamask or etherscan, however, it was never successfully sent.

**Comment by tcraigscott** [comment]
Hmmmm, My transaction shows up in metamask in the sense that the Ether left  my wallet, it also shows up in etherscan. When I say that it does not show up in metamask I mean that if i type in the name again it doesn't say, you have bid xxx amount, it just gives me the option to bid again.

**Comment by frankwit** [comment]
Same here. OSX El Capitan. SSD + 32 GB RAM, ENS app super sluggish yesterday and same problems as OP.
backed-up everything, deleted Mist and support files, re-installed.  imported jsons after merging,  working fine now,  except for the fact that owned domains to be finalized are disappearing from the side bar. Domains cannot be finalized when typed in, mist refuses address lookup, type in a different domain and the retype the domain to be finalized works... strange....

**Comment by belgacoin** [comment]
Hello, I use MEW to reveal my bid. How can I see that I successfully revealed it? All fields are green but that's all I can see right now.
I pasted two tx hashes in etherscan but all I can see is: Unable to locate Transaction entry. 
The tx hash I got after during auction is https://etherscan.io/tx/0x68f5b15c2716c314fbef801472b447902cc6a682423ab12a1ad9849a280c175e

**Comment by frankwit** [comment]
Forget Mist on OSX  El-Capitan, after 2 days of errors I gave up.  Metamask works like a charm.

**Comment by Laramie43ip** [comment]
"If your bid does not show up in metamask or etherscan, however, it was never successfully sent."

This is technically true.  But, I think the issue is: Bid shows up in Etherscan but the registrar DAPP does not cache the name.  It also does not always download names you've successfully bid on.  So, when you got to Reveal, you are unable to do so and funds get burned.  I have many examples, including Etherscan links, .JSON download files after every bid, where this very thing happened to me.

**Comment by yellowthumb** [comment]
Hello I have created a backup file of my bid and have it now in the reveal phase uploaded via the parity browser. But I get displayed "No bids found on file". I also tryed it with with chrome + Metamask, same result. What am I doing wrong?

**Comment by yellowthumb** [comment]
(https://etherscan.io/tx/0x79105ee7d2022de03de8ec7fe20ab24af40691eff2d1c851fa1b80c449e36ef8)
I have not bid on "lanwanprofessional.eth". I have bid on "commercial.eth". But also when I try to reveal the bid for "lanwanprofessional.eth" it shows me "No bids found on file". I dont know whats going on here?!?
There is another bid on business.eth i made. But here (https://etherscan.io/tx/0xe8cf21203e74e884c3ead40ce83f1092cf66210c9e373e41db7a65f5c71ee597)
 it shows me "alymaamh.eth"

**Comment by gregkbarnes** [comment]
Hey @Laramie43ip @yellowthumb and others:

If you have the backup JSON, you can do the reveal using MEW. Fill out the fields manually, but instead of writing your three words, use the **hexSecret** value from your JSON file.

**Comment by apock666** [comment]
Sorry if this fix has been posted, if you cant reveal i tried resetting the Metamask Sever by changing it to test server then back to the main server, seemed to fix the problems .
Cheers

**Comment by Darakaye2** [comment]
Hi,

Can someone help me.  I wasnt able to reveal my bid on ENS and now the ether is locked up to another address.  What can I do?  Its a significant amount of ether. :(

**Comment by Arachnid** [comment]
If the reveal period is already over, there is nothing you can do.

**Comment by gregkbarnes** [comment]
I believe the ether is returned at the end of the 2 years... right?

**Comment by Arachnid** [comment]
No; 19 days after the bid was placed, it can be invalidated by anyone.

**Comment by Darakaye2** [comment]
Nick, how can I do this?

Dara

On May 28, 2017, at 12:03 PM, Nick Johnson <notifications@github.com<mailto:notifications@github.com>> wrote:


No; 19 days after the bid was placed, it can be invalidated by anyone.

**Comment by Darakaye2** [comment]
Nick,

I have my bid amount ,  my masked bid amount plus my 3 words.  I have 51 Eth sitting there.  I could  not get a reveal to work.   Is there a link I can go to and input my info?  What is MEW?  Sorry for all the questions but I'm only a passionate investor in Ethereum.  I am not a coder/programmer. I've been involved since the beginning.  I went through The DAO hack and still am a die hard believer in Eth and the Blockchain.   I just need a little guidance here please.

Dara


On May 28, 2017, at 12:03 PM, Nick Johnson <notifications@github.com<mailto:notifications@github.com>> wrote:


No; 19 days after the bid was placed, it can be invalidated by anyone.

**Comment by Arachnid** [comment]
Is the reveal period already over? If it is, you did not reveal, and you were not the only bidder, your bid is lost - the most you can recover is 0.5%, by revealing late.

**Comment by Darakaye2** [comment]
Yes, Nick. The reveal ended literally as I was trying to reveal.  I wasn't able too.  It's so weird. I have successfully revealed other bids.  This one just wouldn't work.  I came to github immediately for help.

Can you walk me through the process to reveal the .5? It was 51 Eth unfortunately.  I'm so heartbroken over this.  I am immensely grateful for any help.


Dara

On May 30, 2017, at 8:52 AM, Nick Johnson <notifications@github.com<mailto:notifications@github.com>> wrote:


Is the reveal period already over? If it is, you did not reveal, and you were not the only bidder, your bid is lost - the most you can recover is 0.5%, by revealing late.

**Comment by Darakaye2** [comment]
Nick, can you guide me through the .5 recover?

Please and thank you in advance.

Dara

On May 30, 2017, at 8:52 AM, Nick Johnson <notifications@github.com<mailto:notifications@github.com>> wrote:


Is the reveal period already over? If it is, you did not reveal, and you were not the only bidder, your bid is lost - the most you can recover is 0.5%, by revealing late.

**Comment by hackobi** [comment]
> Laramie43ip commented 27 days ago
Also, switching between http and https helps when a bid is missing. But, again, that also is not foolproof. Some confirmed bids do not show up regardless of whether you use http or https.

How do you switch from https to http?

**Comment by dasdqwdqw** [comment]
I have the exact same problem - the reveal button won't show for a bid i made, which i have it in JSTOR. I have tried deleting history and re-importing JSON, MEW doesn't recognize it, and http and https but without any result. I have 30 hours to solve it, can anyone help please! 
P.S. i just saw that the amount i have in the JSTOR as a bid is slightly higher than the bid i made ?!

**Comment by InfiniteProfitSolutions** [comment]
I've got a similar issue. 

It's not recognizing me as a bidder for a term.  I found the Json file with the name, but it's missing this part:
"PendingBids":[{"date":XXXXXXXXXXXXX,"mastersalt":"El Iniahi Neumud 

So it's not recognizing the Json

**Comment by dskvr** [comment]
Have these issues been corrected? I just placed two bids and while preemptively searching for how to reveal bids I came across this. Looking at my various backup files, everything appears to be in place. I even see "pending bids" (which I assume are bids that have not yet been confirmed) moving into "mybids" (which I assume are bigs that have been confirmed) ...  Is there anything I can do now to ensure my backup file has all the required data?

**Comment by alevasa** [comment]
My reveal fase ends in 6hours, I am using Metamask, apparently the page is working well I can push the reveal button confirm everything but it looks like the transaction does not go trough and I find the following message at etherscan "Sorry, we are Unable to locate the Transaction entry." 

https://etherscan.io/tx/0x1033da7fe9206c9a86b601f877bbeb19bb0869b62803bfa8666d118028e7c076

If someone could help me I would appreciate.

**Comment by cfl0ws** [comment]
@alevasa I had a similar problem and think it has to do w/network congestion. Something you could try is to reveal your bid again and set Gwei to 50 or 51. That seems to help cut through the congestion. You can [see how to set Gwei in Metamask here.](https://steemit.com/ethereum/@tomshwom/ethereum-gas-how-it-works)

**Comment by alevasa** [comment]
tks for the tip I did two things, changed network and changed back to
Ethereum Main Net and increased gas price to 51 and it worked :-)

2017-06-22 14:04 GMT-03:00 Chris Remus <notifications@github.com>:

**Comment by cfl0ws** [comment]
@alevasa Glad to hear it :)

**Comment by jjgruen** [comment]
Hello! Can someone help? I have a ENS bid that I cannot reveal. If I try I get an error code "bad information". I have the amount, the secret.
I have tried it manually and also trough myetherwallet.

**Comment by alevasa** [comment]
Try to logout from Ethereum Net and loging again.

Worked for me
Em sáb, 24 de jun de 2017 às 08:14, jjgruen <notifications@github.com>
escreveu:

**Comment by rondior** [comment]
So far I've lost at least 3eth because I couldn't Reveal my bid. I tried for over 2 days. Tried every solution I read and nothing worked.  Im running OSX 10 and using the Ethereum MIST browser.  Very very frustrating to simply burn away money.

**Comment by rondior** [comment]
On many of the ones I lost my money on this is what i see in dev tools

revealed
:
true
revealing
:
false

apparently this means it didnt get revealed because it has burned my money on all of them.  Now up to 5 ETH lost.

**Comment by Laramie43ip** [comment]
Haven't used Mist but had the same experiences with MetaMask.  I believe
this was due to memory leaks, using Metamask in a Chrome browser running OS
X.

Switched to MyEtherWallet several months back and those problems
disappeared.  The one problem that occasionally occurs: If you bid on an
auction, sometimes it doesn't "launch."  This happens rarely but it does
happen.  Always save your reveal hash so you can "prove" that you bid, even
if you can't prove what name you bid on.  The solution that I've found to
this is either: (a) wait for someone else to initiate an auction on that
same name, then you'll be able to reveal your bid (even if it's more than 3
days old when the reveal period starts); or (b) relaunch the auction
yourself for .01 Eth, then reveal your original bid and the "relaunch" bid
of .01.  Both always have worked.  This has happened on at least a dozen
occasions and I've always been able to reveal even the bid that didn't
launch the auction.

Wish I could help with the burned Eth.  Unfortunately, I'm in the same boat
for the Eth that was burned until I transitioned away from Metamask.

**Comment by Darakaye2** [comment]
Guys- can I get my burned Eth back?  Someone said challenge it?  Can someone elaborate. 50eth gone. Please help.  It's been over a month now. Reveal wouldn't work.

Dara K. Hudson
ÜberMed, Inc.
dara@ubermedical.org<mailto:dara@ubermedical.org>
Cell-630-631-6043
Chicago

On Jul 1, 2017, at 7:02 PM, Laramie43ip <notifications@github.com<mailto:notifications@github.com>> wrote:

Haven't used Mist but had the same experiences with MetaMask. I believe
this was due to memory leaks, using Metamask in a Chrome browser running OS
X.

Switched to MyEtherWallet several months back and those problems
disappeared. The one problem that occasionally occurs: If you bid on an
auction, sometimes it doesn't "launch." This happens rarely but it does
happen. Always save your reveal hash so you can "prove" that you bid, even
if you can't prove what name you bid on. The solution that I've found to
this is either: (a) wait for someone else to initiate an auction on that
same name, then you'll be able to reveal your bid (even if it's more than 3
days old when the reveal period starts); or (b) relaunch the auction
yourself for .01 Eth, then reveal your original bid and the "relaunch" bid
of .01. Both always have worked. This has happened on at least a dozen
occasions and I've always been able to reveal even the bid that didn't
launch the auction.

Wish I could help with the burned Eth. Unfortunately, I'm in the same boat
for the Eth that was burned until I transitioned away from Metamask.
